gateway-shell/src/main/java/org/apache/knox/gateway/shell/table/KnoxShellTableBuilder.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.shell.table;
+
+
+public class KnoxShellTableBuilder {
+  protected String title;
+
+  public KnoxShellTableBuilder title(String title) {
+    this.title = title;
+    return this;
+  }
+
+  public CSVKnoxShellTableBuilder csv() {
+    return new CSVKnoxShellTableBuilder();
```

Comment:
What would be kinda cool is a service loader approach here (like how other parts of Knox work) to avoid explicit methods. Not sure how you would do that though with the methods on KnoxShellTableBuilder.

Reply:
I did intend to add serviceLoader support like that from the beginning.
I think that an additional method could be added to the KnoxShellBuilder to accept the name.
KnoxShellTable.builder().name("csv").withHeaders().url("http://...")